modules/apps/cookies/cookies-banner-web/src/main/resources/META-INF/resources/js/toggleThirdPartyCookies.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {COOKIE_TYPES, checkConsent} from 'frontend-js-web';
+
+function flipThirdPartyCookie(type) {
+	const selector = type ? `[data-third-party-cookie="${type}"]` : '[data-third-party-cookie]'
+
+	document.querySelectorAll(selector).forEach(
+		element => {
+			element.removeAttribute('data-third-party-cookie')
+
+			switch (element.tagName) {
+				case 'SCRIPT': {
+					const newScript = element.cloneNode();
+
+					newScript.type = 'text/javascript';
+
+					element.replaceWith(newScript);
+					break;
+				}
+				case 'LINK': {
+					const newLink = element.cloneNode();
+
+					newLink.href = element.dataset['href'];
+					newLink.removeAttribute('data-href');
+
+					element.replaceWith(newLink);
+					break;
+				}
+				case 'EMBED':
+				case 'IFRAME':
+				case 'IMG':
+					element.src = element.dataset['src'];
+					element.removeAttribute('data-src');
+					break;
+				default:
+					// eslint-disable-next-line no-console
+					console.log('Error: ', element.tagName);
+			}
+		}
+	)
+}
+
+export default function toggleThirdPartyCookies() {
+	Object.values(COOKIE_TYPES).forEach(
+		type => {
+			if (checkConsent(type)) {
+				flipThirdPartyCookie(type)
+			}
+		}
+	)
+}
```

Comment:
Some thoughts about timing and `toggleThirdPartyCookies()`:
* May we find race conditions depending on when it runs vs when the CE is loaded? More than race conditions, I mean the situation where either a) CE could run then browser loaded the `<img>`, `<script>` etc. before we have a chance to act, or b) the other way round: cookie toggling happens before CE script runs. 
* Current solution prevents a) by having `data-src` attribute, which is the way we have to enforce. If we go that route, using this `data-src` must be part of the contract to include compliant content.
* In general, to cover b) any dynamically added element with `data-src` attribute is likely to be target of this logic, so the contract about using `data-src` must be obeyed also. In this regard, we might consider a mutation observer to ensure proper toggles take place on those dynamically added things.
* Any change in the preferences requires triggering this. IIRC, if user said yes to some type of cookie then rejected, whatever was created while consent was given can remain in the user's device